integration-cli/docker_cli_wait_test.go

```diff
@@ -120,3 +121,43 @@ func TestWaitBlockedExitRandom(t *testing.T) {
 
 	logDone("wait - blocking wait with random exit code")
 }
+
+func TestWaitKilledContainerNoWait(t *testing.T) {
+	defer deleteAllContainers()
+
+	runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "/bin/cat")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		t.Fatal(out, err)
+	}
+	containerID := strings.TrimSpace(out)
+
+	killCmd := exec.Command(dockerBinary, "kill", containerID)
+	out, _, err = runCommandWithOutput(killCmd)
+	if err != nil {
+		t.Fatal(out, err)
+	}
+
+	done := make(chan struct{})
+	var wg sync.WaitGroup
```

Comment:
https://github.com/docker/docker/pull/12109#discussion_r27812654
also here https://github.com/docker/docker/issues/11529

I think it make sense and started to use it whenever possible, if it's a bad pattern or unnecessary I'll remove it..